kroxylicious-kms-provider-aws-kms/src/main/java/io/kroxylicious/kms/provider/aws/kms/AwsKms.java

```diff
@@ -85,8 +83,11 @@ public class AwsKms implements Kms<String, AwsKmsEdek> {
      */
     private final URI awsUrl;
 
-    AwsKms(@NonNull URI awsUrl, @NonNull CredentialsProvider credentialsProvider, @NonNull String region, @NonNull Duration timeout,
-           @NonNull UnaryOperator<Builder> tlsConfigurator) {
+    AwsKms(URI awsUrl,
+           CredentialsProvider credentialsProvider,
+           String region,
+           Duration timeout,
+           UnaryOperator<Builder> tlsConfigurator) {
```

Comment:
timeout stands out because it is absent from parameter checked with Objects.requireNonNull).  java.net.http.HttpClient.Builder#connectTimeout actually enforces the not-null, so suggest adding an `Objects.requireNonNull(timeout);` to the code.
```suggestion
           UnaryOperator<Builder> tlsConfigurator) {
           Objects.requireNonNull(timeout);
```